config/jenkins-jobs_infra.ci.jenkins.io.yaml

```
@@ -441,7 +441,7 @@ jobsDefinition:
           algolia-plugins-write-key:
             secret: "${ALGOLIA_PLUGINS_JENKINS_IO_WRITE_KEY}"
             description: "Algolia credentials to write data for plugin site"
-          infraci-pluginsjenkinsio-fileshare-service-principal-writer:
+          infraci_pluginsjenkinsio_fileshare_serviceprincipal_writer:
```

- This YAML key is not using the same convention as the other keys (above and below) with this change, which defeats the purpose of the change itself (avoid confusion)
- Changing the key id means breaking the associated pipeline (ref. https://github.com/jenkins-infra/plugin-site/blob/7d3ce54af398b75553e96cdbc859fa0b0960f4a6/Jenkinsfile#L132) until it is updated.


=> As much as the other renaming make sense in the context of terraform itself, this change here is not wanted unless a compelling reason

---

```suggestion
          infraci-pluginsjenkinsio-fileshare-service-principal-writer:
            kind: azure-serviceprincipal
            azureEnvironmentName: "Azure"
            clientId: "${INFRACI_PLUGINSJENKINSIO_FILESHARE_SERVICE_PRINCIPAL_WRITER_CLIENT_ID}"
            clientSecret: "${INFRACI_PLUGINSJENKINSIO_FILESHARE_SERVICE_PRINCIPAL_WRITER_PASSWORD}"
            description: "plugins.jenkins.io File Share Service Principal Writer"
            subscriptionId: "${JENKINSINFRA_AZURE_PRIMARY_SUBSCRIPTION_ID}"
            tenant: "${JENKINSINFRA_AZURE_TENANT_ID}"
          infraci_pluginsjenkinsio_fileshare_serviceprincipal_writer:
```
Does this change duplicating the credentials until the plugin-site pipeline is updated would be acceptable?

Having the same name format everywhere would be nice for consistency.

---

All credentials were concerned, hence https://github.com/jenkins-infra/kubernetes-management/pull/5356/commits/bebacc4caba04281edf30560d65c20ec47892783

WDYT?

---

> Does this change duplicating the credentials with a new name until the plugin-site pipeline is updated would be acceptable?

In term of "not breaking the pipeline": yes

> Having the same name format everywhere would be nice for consistency.

In theory: yes. In practice: 
- Not if it create inconsistencies everywhere else (like this PR unless we change ALL credentials inb this file with the associated effort to update all pipeline) => not worth the effort
- And not if the convention become a hard process constraining everything: if terraform objects naming convention constrains the Jenkins credential IDs specified by the Jenkins pipeline (unrelated areas, and potentially unrelated actors) => coupling is the worst we could make in an infra-as-code world.